test/org/javarosa/xpath/expr/XPathPathExprCurrentItemsetNodesetTest.java

```diff
@@ -0,0 +1,84 @@
+import org.javarosa.core.model.data.SelectOneData;
+import org.javarosa.core.model.data.StringData;
+import org.javarosa.core.model.data.helper.Selection;
+import org.javarosa.core.model.instance.InstanceInitializationFactory;
+import org.javarosa.core.test.FormParseInit;
+import org.javarosa.form.api.FormEntryController;
+import org.javarosa.form.api.FormEntryModel;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.junit.Before;
+import org.junit.Test;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.javarosa.test.utils.ResourcePathHelper.r;
+
+public class XPathPathExprCurrentItemsetNodesetTest {
+    private FormDef formDef;
+    private FormEntryController formEntryController;
+
+    @Before
+    public void setUp() {
+        FormParseInit fpi = new FormParseInit(r("relative-current-ref-itemset-nodeset.xml"));
+        formDef = fpi.getFormDef();
+        formDef.initialize(true, new InstanceInitializationFactory());
+        FormEntryModel formEntryModel = new FormEntryModel(formDef);
+        formEntryController = new FormEntryController(formEntryModel);
+    }
+
+    /**
+     * current() in an itemset nodeset expression should refer to the select node. This is verified by building an
+     * itemset from a repeat which is a sibling of the select.
+     */
+    @Test
+    public void current_in_itemset_nodeset_should_refer_to_node() {
+        // don't know how to jump to repeat directly so jump to following question and step backwards
+        formEntryController.jumpToFirstQuestionWithName("selected_person");
+        formEntryController.stepToPreviousEvent(); // repeat
+        formEntryController.descendIntoRepeat(0);
+        formEntryController.stepToNextEvent(); // person
+
+        StringData nameValue = new StringData("Bob");
+        formEntryController.answerQuestion(nameValue, true);
+
+        formEntryController.stepToNextEvent();
+        formEntryController.descendIntoNewRepeat();
+        formEntryController.stepToNextEvent(); // person
+
+        StringData nameValue2 = new StringData("Janet");
+        formEntryController.answerQuestion(nameValue2, true);
+
+        formEntryController.jumpToFirstQuestionWithName("selected_person");
+        FormEntryPrompt personPrompt = formEntryController.getModel().getQuestionPrompt();
+        personPrompt.getAnswerValue();
+        ItemsetBinding dynamicChoices = personPrompt.getQuestion().getDynamicChoices();
+
+        SelectChoice personChoice = dynamicChoices.getChoices().get(1);
+        assertEquals("Janet", personPrompt.getSelectChoiceText(personChoice));
+        SelectOneData personSelection = new SelectOneData(new Selection(personChoice));
+        formEntryController.answerQuestion(personSelection, true);
+
+        SelectOneData personSelectionValue = (SelectOneData) formDef.getFirstDescendantWithName("selected_person")
+                .getValue();
+        assertEquals("2", personSelectionValue.getDisplayText());
```

Comment:
Why is 2 the display text value for "Janet"? Where does that 2 come from?

Reply:
Never mind! I figured out this is the count of people :)

Reply:
For this specific test, I feel it's really important to make sure both value and label are correctly populated because those come from relative expressions built off of the nodeset path. This first assert confirms that the value was correctly resolved. The assert before it verifies that the label was correctly resolved.  

Like with the other related tests, I'd be ok with doing both asserts where the first one is and dropping the "round trip" aspect of saving a value and reading it back.

As an additional note, I also think it's important to not check the first choice here. For example, using `position()` rather than `position(..)` in the form would mean all choices have value 1 and that could hide another problem. So in general, checking something where an index actually has to change seems like a good idea.